File: lab-1/domain/podcaster.py
from domain.performer import Performer


class Podcaster(Performer):
    def __init__(self, name, debut_date, dob, topic):
        super().__init__(name)
        self.debut_date = debut_date
        self.dob = dob
        self.topic = topic
        self.genre = "Podcast"

    def introduce(self):
        print(f"Hello, I am {self.name}, a podcast host. I have been active since {self.debut_date}.")
        print(f"Usually, I talk about {self.topic}.")

    def perform(self):
        print(f"{self.name} is currently hosting a podcast episode.")

File: src/main/kotlin/performer/implementations/Podcaster.kt
package performer.implementations
import performer.Performer

class Podcaster(
        override val name: String,
        val debutDate: String,
        val dob: String,
        val topic: String
) : Performer {
    val genre = "Podcast"

    override fun introduce() {
        println("Hello, I am $name, a podcast host. I have been active since $debutDate.")
        println("Usually, I talk about $topic.")
    }

    fun perform() {
        println("$name is currently hosting a podcast episode.")
    }
}